### PR TITLE
Feat/node group integration

### PR DESCRIPTION
**Pull request by jvitor03**
Add endpoint to read NodeGroups metadata

### CODE REVIEW

**Review by L30Bola — CHANGES_REQUESTED** [review]
On `internal/controller/nodeGroup_controller.go`  _NodeGroupByClusterHandler_ is using the "old" way to handle errors on the `GetCluster` block. That should be updated.

**Review by L30Bola — COMMENTED** [review]
Better explained on below conversation.

File: internal/k8s/nodeGroup.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("could not Marshal machinepool response: %v", err)
+	}
+
+	err = json.Unmarshal(machinePoolRawJson, &machinePool)
+	if err != nil {
+		return nil, fmt.Errorf("could not Unmarshal machinepool JSON into clusterAPI list: %v", err)
+	}
+
+	//err := sanitizeMachinePool(&machinePool)
```
Comment (L30Bola) [inline internal/k8s/nodeGroup.go]: Why is this block of code commented?

Comment (jvitor03) [reply internal/k8s/nodeGroup.go]: It should be removed.   
CAPI sanitizes the machinepool/machinedeployment, we should not need to confirm what's already confirmed.

**Review by L30Bola — COMMENTED** [review]
Besides the aforementioned changes, I think it's all good.

**Comment by jvitor03** [comment]
Done requested changes

**Review by L30Bola — APPROVED** [review]
Now, it's all set. I just adjusted some method descriptions.

**Pull request merged**